File: web/src/apis/topology.ts
import { httpClient } from './request'

export interface TopologyConfig {
  id: number | string
  projectId: number | string
  hostId: string
  process: string
  createdAt: string
  updatedAt: string
}

export async function listTopologyConfig(projectId: number | string): Promise<TopologyConfig[]> {
  const res = await httpClient.get<any[]>('/api/topology', { project_id: projectId })
  const list = res.data ?? []
  // 后端下划线 → 前端驼峰映射
  return (Array.isArray(list) ? list : []).map((it: any) => ({
    id: it.id,
    projectId: it.project_id,
    hostId: it.host_id ?? '',
    process: it.process ?? '',
    createdAt: it.created_at ?? '',
    updatedAt: it.updated_at ?? '',
  }))
}

File: server/server/service/topology/topology.go
package topology

// TopologyConfigDTO 拓扑配置实体
type TopologyConfigDTO struct {
	ID        int64  `json:"id"`
	ProjectID int    `json:"project_id"`
	HostID    string `json:"host_id"`
	Process   string `json:"process"`
	CreatedAt string `json:"created_at"`
	UpdatedAt string `json:"updated_at"`
}
